File: python/raindrops/raindrops.py

def convert(number):
    logic = {3: 'Pling', 5: 'Plang', 7: 'Plong'}
    r = ''
    for k in logic:
        if number % k == 0:
            r += logic[k]
    return str(number) if r == '' else r

File: r/raindrops/raindrops.R
raindrops <- function(number) {
  divisors <- c(3, 5, 7)
  sounds <- c("Pling", "Plang", "Plong")
  r <- paste(sounds[number %% divisors ==0],collapse="")
  if (r == "") {
    return(paste0(number))
  }
  return(r)
}
